internal/pkg/runtime/engines/singularity/prepare_linux.go

```diff
@@ -720,6 +721,20 @@ func (e *EngineOperations) PrepareConfig(starterConfig *starter.Config) error {
 		starterConfig.SetCapabilities(capabilities.Ambient, e.EngineConfig.OciConfig.Process.Capabilities.Ambient)
 	}
 
+	// check if there is a controlling terminal
```

Comment:
```suggestion
	// obtain the process group ID of the associated controlling
	// terminal (if there's one).
```

This little bit here could be its own function, just for clarity's sake.